frontend/test/metabase/scenarios/question/native_subquery.cy.spec.js

```diff
@@ -0,0 +1,76 @@
+import { signInAsAdmin, restore, signIn } from "__support__/cypress";
+
+describe("scenarios > question > native subquery", () => {
+  before(restore);
+  beforeEach(signInAsAdmin);
+
+  it("should allow a user with no data access to execute a native subquery", () => {
+    // Create the initial SQL question and followup nested question
+    cy.request("POST", "/api/card", {
+      name: "People in WA",
+      dataset_query: {
+        type: "native",
+        native: {
+          query: "select * from PEOPLE where STATE = 'WA'",
+        },
+        database: 1,
+      },
+      display: "table",
+      description: null,
+      visualization_settings: {},
+      collection_id: null,
+      result_metadata: null,
+      metadata_checksum: null,
+    })
+      .then(response => {
+        cy.wrap(response.body.id).as("nestedQuestionId");
+        const tagID = `#${response.body.id}`;
+
+        cy.request("POST", "/api/card", {
+          name: "Count of People in WA",
+          dataset_query: {
+            type: "native",
+            native: {
+              query: `select COUNT(*) from {{#${response.body.id}}}`,
```

Comment:
You still need SQL permissions to run SQL queries. The test is failing because the user doesn't have SQL permissions -- not because the user doesn't have permissions for the source query

Reply:
Interesting...this is a little confusing to me. Because that same user _can_ run the original SQL question. 

Shouldn't a user with collection access be able to run anything in there regardless of if they have SQL access or not?

Reply:
Collection permissions don't apply to ad-hoc queries, since they're not in any Collections. Collection permissions only apply to saved Questions. Data permissions apply to ad-hoc questions; to run an ad-hoc SQL query you need to have SQL permissions for the database in question

Reply:
if the original SQL question is *saved* then Collection permissions apply to it -- the important distinction is that saved Questions require permissions for their parent Collection while ad-hoc queries require data permissions for the database/table in question.

Reply:
Yea, in the test it’s not an ad-hoc SQL query, it’s a saved question. So the user with “collection only” access should be able to run it right?

Reply:
sorry @dacort, you're totally right, I read this Cypress test wrong and misunderstood what it's testing. I'm still a little confused as to why the error is `Expected to find content: '41' but never did.` when it looks like it tries the nested query with the no-data-perms user *first* and the top-level question second. If it couldn't access the nested question, wouldn't it fail trying to find `Showing 41 rows`?

Either way I will take another look at this tomorrow.